src/main/scala/it/pagopa/interop/tenantmanagement/model/persistence/TenantPersistentBehavior.scala

```diff
@@ -91,6 +91,20 @@ object TenantPersistentBehavior {
           Effect.persist(SelfcareMappingDeleted(selfcareId)).thenReply(replyTo)(_ => success(()))
         else Effect.reply(replyTo)(error[Unit](SelfcareIdNotMapped(selfcareId)))
 
+      case DeleteTenantMail(tenantId, mailId, replyTo) =>
+        if (state.tenants.contains(tenantId.toString))
+          Effect.persist(TenantMailDeleted(tenantId, mailId)).thenReply(replyTo)(_ => success(()))
+        else Effect.reply(replyTo)(error[Unit](TenantNotFound(tenantId.toString)))
```

Comment:
@beetlecrunch should we return an error if the mail id does not exist?

Reply:
@galales This should be a rare case, but let us add a specific error for this case (MailNotFound).
cc @nttdata-rtorsoli

Reply:
Just Done and pushed